elyra/pipeline/component.py

```diff
@@ -202,7 +204,20 @@ def __init__(self, id: str, name: str, description: Optional[str], source_type:
         self._runtime = runtime
         self._op = op
         self._properties = properties
-        self._extension = extension
+
+        # Error checking for extensions & filehandler_parameter_ref depends on the component source type
+        if self._source_type == "elyra":
+            if not extensions:
+                raise ValueError("Invalid generic component: Missing field 'extensions'.")
+            filehandler_parameter_ref = "filename"
+        else:
+            if extensions and not filehandler_parameter_ref:
+                raise ValueError("Invalid component: A component for which file extensions are \
+                                 supplied must include the parameter_ref of the first parameter \
+                                 that is associated with that extensinon.")
```

Comment:
This feels a little strict to me.  I think we'll always be passing the extensions for our 'elyra' components and, someday, new ones may not use extensions.  I also think a missing `filehandler_parameter_ref` is more of a warning when there are extensions specified than a fatal error.  The component should still function, just that drag and drop can't be supported.  However, even when specified, the custom component's drag/drop behavior will likely be superseded by the drag/drop behavior of generic components unless the specified extensions are different.   Therefore, I think reducing this if/else to a warning message if extensions are set and fh_p_r is not set is more appropriate.  Perhaps the warning could read something like:
 
 "Component '{op}' specifies extensions '{extensions}' but no 'filehandler_parameter_ref' value and cannot participate in drag and drop functionality as a result."

Reply:
Great point!